components/script/dom/file.rs

```diff
@@ -18,14 +20,15 @@ pub struct File {
 }
 
 impl File {
-    fn new_inherited(_file_bits: &Blob, name: DOMString) -> File {
+    fn new_inherited(file_bits: &Blob, name: DOMString) -> File {
+       let mime_type = guess_mime_type(Path::new(&*name));
```

Comment:
Why are we guessing the type here? The spec doesn't mention this, the type is the empty string unless explicitly provided by the user through the FilePropertyBag/BlobPropertyBag.